File: tests/fixtures/all_hooks_custom.py
import pandas as pd

prediction_value = None


def init(**kwargs):
    global prediction_value
    prediction_value = 1


def load_model(input_dir):
    global prediction_value
    prediction_value += 1
    return "dummy"


def transform(data, model):
    global prediction_value
    prediction_value += 1
    return data


def score(data, model, **kwargs):
    global prediction_value
    prediction_value += 1
    predictions = pd.DataFrame(
        [prediction_value for _ in range(data.shape[0])], columns=["Predictions"]
    )
    return predictions


def post_process(predictions, model):
    return predictions + 1

File: tests/fixtures/all_hooks_custom.R
prediction_value <- NaN


init <- function(...) {
    prediction_value <<- 1
}

load_model <- function(input_dir) {
    prediction_value <<- prediction_value + 1
    "dummy"
}

transform <- function(data, model) {
    prediction_value <<- prediction_value + 1
    data
}

score <- function(data, model, ...) {
    prediction_value <<- prediction_value + 1
    predictions = data.frame(matrix(prediction_value, ncol = 1, nrow = nrow(data)))
    names(predictions) <- c("Predictions")
    predictions
}

post_process <- function(predictions, model) {
    predictions + 1
}
